src/views/RequestQueueView.js

```diff
@@ -62,10 +62,19 @@ class RequestQueueView extends React.Component {
   };
 
   static getDerivedStateFromProps(props, state) {
-    const { data: { requests } } = props;
+    const { data: {
+      notYetFilledRequests,
+      inProgressRequests,
+    } } = props;
+
+    const requestsPositionsForReorder = notYetFilledRequests.map(r => r.position);
 
     if (!state.reorder) {
-      return { requests };
+      return {
+        notYetFilledRequests,
+        inProgressRequests,
+        requestsPositionsForReorder,
```

Comment:
Discussed => move this logic to constructor and componentDidUpdate

Reply:
Done